### PR TITLE
Fix ImgLabeling for various types of index images

### PR DESCRIPTION
**Pull request by maarzt**
The accessors for ImgLabeling are build with the assumption that
RandomAccess.get() and Cursor.get() of the index image will always return
the same object. This constraint holds for ArrayImg and CellImg, but fails
for other RandomAccessibleIntervals like:
* ListImg,
* ConvertedRandomAccessilbeInterval,
* StackView

This commit removes the assumption, and therefor fixes ImgLabeling for the
previously mentioned RandomAccessibleIntervals.

### CODE REVIEW

File: src/main/java/net/imglib2/roi/labeling/LabelingType.java [hunk 1]
```diff
@@ -83,6 +83,10 @@ protected LabelingType( final IntegerType< ? > type, final LabelingMapping< T >
 		this.generation = modCount;
 	}
 
+	public void setType( final IntegerType< ? > type) {
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/roi/labeling/LabelingType.java]: Should not be public

**Comment by tpietzsch** [comment]
Can you please add benchmarks to show the impact this has on `ArrayImg` backed `ImgLabeling`?
I clearly see the need to make it work for all RandomAccessibleIntervals, but it would be good to know the performance impact...

**Comment by maarzt** [comment]
The current code must be considered broken. It's for example not possible to use a cache image backed ImgLabeling, which is a big issue. I strongly doubt the need for benchmarking broken code against working code. But here are the results:

Broken code:
```
Benchmark                                    Mode  Cnt      Score      Error  Units
ImgLabelingBenchmark.benchmarkCursor        thrpt    8  34039.611 ± 6137.750  ops/s
ImgLabelingBenchmark.benchmarkRandomAccess  thrpt    8  14425.188 ± 1376.425  ops/s
```
This PR:
```
Benchmark                                    Mode  Cnt      Score      Error  Units
ImgLabelingBenchmark.benchmarkCursor        thrpt    8  28140.803 ± 6749.818  ops/s
ImgLabelingBenchmark.benchmarkRandomAccess  thrpt    8  13572.857 ±  767.816  ops/s
```
10% performance loss, in the specific scientific test cases. The performance loss in real life is probably negligible.

**Comment by maarzt** [comment]
I changed the benchmark a tiny bit, to now use a larger image, and therefor do a more precise measurement.
Conclusion:
*  No performance impact on the `RandomAccess`
* `Cursor` is 10% slower in the test case, but this is very likely not relevant for real life code.

**Comment by ctrueden** [comment]
Wow, this has been open for quite a long time. It had conflicts, so I rebased it over the latest master and force-pushed.

**Comment by imagejan** [comment]
What's the status of this one? Can it be rebased and merged?

**Comment by maarzt** [comment]
I think it can be merged. @tpietzsch What do you think?
The performance tests are done. 10% slower for ArrayImg cursor.

**Pull request merged**